### ISSUE TITLE
1.x migration tutorial - Restart required when editing config files

### ISSUE THREAD

**Issue by bgilmer77** [openhab/openhab-docs#409]
Issue:  The migration tutorial [here](http://docs.openhab.org/tutorials/migration.html) fails to mention that text-based config files are read-only, and that a restart is required in order for changes to go into effect.  This is particularly challenging because, when you make a change to a text-based rules file, a note appears in the log saying that the rules file has just been read.  Hum.  Read, but not put into effect!  I spent hours troubleshooting before I figured out that I had to stop and start OH2 in order to get rules to go into effect.

Suggested resolution:  Add the following to the 'paperUI Approach' section of the migration tutorial-

In order to migrate from OH1 to OH2 successfully, it is important to understand that the basic approach to configuration is changing.  In OH1, configuration was done through text files.  While text files are still an option in OH2, most configuration may be done quickly and intuitively using the paperUI.

There are some important things you should know about this.  
- paperUI configurations are stored in a jasonDB database located at /var/lib/openhab2/jsondb.
- paperUI configurations are *not* written out to configuration files.  They are only stored in the database.
- You can use text files alongside paperUI configurations.  
- Text configuration files  are stored in the appropriate subdirectory under /etc/openhab2/.
- Text configuration files are read-only.  This may result in unexpected behavior.  If you alter a running openhab2 configuration using paperUI, the change is written to jsondb.  The paperUI does *not* alter the original text-based configuration file.  This can be confusing because later, the running system configuration will not be what is indicated by the text files.
- The ability to make and change rules through the paperUI are being added, but in the meantime, rules can be created as text files.  When you make a change to a .rules file, you will see a message similar to the following in the /var/log/openhab2/openhab.log file:

'Refreshing model 'your_rule.rules'

N.B. however, that your rule will *not* go into effect until you restart openhab2 using the following commands:

systemctl stop openhab2.service
systemctl start openhab2.service

**Comment by ThomDietrich** [comment]
Hello @bgilmer77,
thanks for this long description and the offered addition!

While I like your list and would very much like to include it in the tutorial (pinging @rkoshak for that) I can not agree with your main argument. Configuration files are always loaded and taken into effect immediately. I do however not understand what you are trying to do/achieve exactly, especially related to entry 5 of your list. Related to that I don't get the relevance of entry number 6 - rules can be created in rule files as everyone knows, other options are added step by step. Do we need to emphasize this fact?

Best! Thomas

**Comment by bgilmer77** [comment]
> Configuration files are always loaded and taken into effect immediately.
Not on my system.  But you were kind enough to point out that a number of changes have been made in the past few months.  I am on the stable distribution branch.  You suggested that I try the snapshot branch, which I will now do.

When I go to the snapshot branch, if I still have problems, I will open another issue.  Thank you.

**Comment by ThomDietrich** [comment]
Yes, please let us know of your experience.
Let's leave the issue open as the other points on your list are still valid. Let's wait for @rkoshak 's opinion. In general if you think something is missing you can also open a PR right away.

**Comment by bgilmer77** [comment]
Okay - question.  Will switching to the 'testing' or 'unstable' apt distro get me to the snapshot, or should I just pull down the zip?  Also, given this situation, is it best to wipe the entire existing installation (apt-get purge openhab2), or should I install the snapshot over the existing installation?

**Comment by bgilmer77** [comment]
Reporting back.  Yes - installing the snapshot from cloudbees fixes my issue.  Changes in rules go into effect immediately.

**Comment by rkoshak** [comment]
Some quick comments:

1. No problem with this. There was no JASONDB when the migration tutorial was first written (I think). This should be added somewhere, perhaps the Necessary Changes section where the other list of things that are different are listed.

2. Yes, however not all configurations go to JSONDB. For example, bindings installed through PaperUI go ??? (somewhere in cache presumably). I do not know the full list of what goes where when configured through PaperUI and if we are going to document this, it should be thoroughly documented.

I'll also argue that it probably belongs to be documented somewhere in the user manual, not this one tutorial as all users would be interested in knowing this information. I don't know if there is an article already in place where it makes sense to put it. If not then maybe an entry in Advanced is appropriate. The tutorial should link to that.

3. Yes but the text file takes precedence in all cases. For example, if you list addons in addons.cfg, if you try to install an addon using PaperUI, it will immediately be uninstalled so it matches the list in addons.cfg. The same should also go for entries in services but I'm not positive about that (e.g. if I configure Nest through PaperUI and then create and edit a nest.cfg, does nest.cfg take precedence? I remember having this question back when I wrote the tutorial and never getting an answer.

4. This is already in the tutorial:

> In your configurations folder for openHAB 2 (/etc/openhab2 on apt-get installed instances)

The first line of the Text Based section.

5. Better worded as "Text configs and PaperUI configuration changes are managed separately. Changes made in PaperUI do not get reflected in the text config files. The Text config files will always take precedence over PaperUI changes. Entries in text config files (e.g. Items, Things etc) will appear in PaperUI but not be modifiable in PaperUI."

6. This is incorrect as you and Thom already discussed. The expected and correct behavior is that any changes go into immediate effect when loaded.

As I went through some section of the Tutorial I notice a few other things that need to be updated. For example:

> One thing to be aware of is any configuration done in PaperUI or Habmin gets saved to a non-human readable database. And since none of the administration UIs are currently capable of doing everything, you will end up with a mix of database and text based configuration. This is why we recommend not using PaperUI for those migrating to openHAB 2 from openHAB 1.x.

This is no longer true. The DB is now JSONDB which is human editable and readable, at least for those things that go into JSAONDB. I'm also not sure that the community still recommends sticking with text files for most everything for those migrating from OH 1.
